holidays/countries/hongkong.py

```diff
@@ -45,173 +40,379 @@ class HongKong(
     StaticHolidays,
 ):
     """
-    https://en.wikipedia.org/wiki/Public_holidays_in_Hong_Kong
-    Holidays for 2007–2023 (government source):
-    https://www.gov.hk/en/about/abouthk/holiday/index.htm
+    References:
+        - `English Wikipedia <https://en.wikipedia.org/wiki/Public_holidays_in_Hong_Kong>`_
+        - `Chinese Wikipedia <https://zh.wikipedia.org/wiki/香港節日與公眾假期>`_
+    Statutory Holidays:
+        - `Section 39 of Cap. 57 Employment Ordinance <https://www.elegislation.gov.hk/hk/cap57!en-zh-Hant-HK?INDEX_CS=N&xpid=ID_1438403463460_002>`_  # noqa: E501
+        - `Holidays for 2010-2024 <https://www.labour.gov.hk/eng/news/holidays_list.htm>`_
+    General Holidays:
+        - `Cap. 149 General Holidays Ordinance <https://www.elegislation.gov.hk/hk/cap149!en-zh-Hant-HK?INDEX_CS=N>`_  # noqa: E501
+        - `Holidays for 2007–2024 <https://www.gov.hk/en/about/abouthk/holiday/index.htm>`_
     """
 
     country = "HK"
-    observed_label = "The day following %s"
+    # %s（慶祝）.
+    observed_label = "%s (observed)"
+    supported_categories = (OPTIONAL, PUBLIC)
 
-    def __init__(self, *args, **kwargs):
+    def __init__(self, preferred_discretionary_holidays=(CHRISTMAS,), *args, **kwargs):
```

Comment:
I think the best approach to make this extensible is a separate mixin, e.g. `class PreferredDiscretionaryHolidays` which would be responsible for related logic. In this way we would be able to reuse it for other countries if needed.

Reply:
I've just pushed some code implementing my idea. Please review and add your feedback/comments if any.

Thanks!

Reply:
LGTM!